### PR TITLE
new rebuild script to handle ffmpeg repo refreshing, patching, config…

### PR DESCRIPTION
**Pull request by PierceLBrooks**
…uration, compilation, and installation

### CODE REVIEW

File: rebuild.rb [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+      end
+      Dir.chdir(pwd)
+    end
+  end
+  return 0
+end
+
+def configure(pwd, target)
+  Dir.chdir(File.join(pwd, target))
+  command = ["sudo", "./configure", "--enable-shared", "--disable-static", "--enable-openssl", "--prefix=/usr/local/"]
```
Comment (stack) [inline rebuild.rb]: Only `make install` should be run with `sudo`.

File: rebuild.rb [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    Dir.chdir(pwd)
+    return -1
+  end
+  Dir.chdir(pwd)
+  return 0
+end
+
+def make(pwd, target)
+  Dir.chdir(File.join(pwd, target))
+  command = ["sudo", "make"]
```
Comment (stack) [inline rebuild.rb]: Only `make install` should be run with `sudo`.

**Review by stack — CHANGES_REQUESTED** [review]
I think this looks good if you take out the unneeded calls to `sudo`.

**Comment by PierceLBrooks** [comment]
@stack:
[https://github.com/SquirrelsLLC/ffmpeg-patches/pull/1/commits/7f5683efffa0ea662dbec0bd96d5b50cb664ca5b](url) should resolve your request

**Comment by stack** [comment]
Two thoughts on this:

1. Should we be doing the `sudo apt-get build-dep libavcodec ...` stuff in here?
2. Should the first `make` call have the `-j` option enabled, so that it does parallel builds?

**Comment by PierceLBrooks** [comment]
I can add the `sudo apt-get build-dep` calls and build parallelization parameter to `make` now.

**Comment by stack** [comment]
Looks good!

**Pull request merged**